### PR TITLE
QA Observation Bugfix/FOUR-12775: The upload images control does not validate the format of the files.

### PR DESCRIPTION
**Pull request by CarliPinell**
## Solution
- Only PNG and JPG extensions are allowed in image container.

## How to Test
- Open a Process
- Click on PUBLISH button
- Upload or Drag Images into container
- ONLY PNG and JPG files are allowed.

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-12787

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.
ci:deploy
ci:next

### CODE REVIEW

File: resources/js/components/shared/ModalSaveVersion.vue [hunk 1]
```diff
@@ -382,6 +385,17 @@ export default {
...
+    },
+    /**
+     * Validate image extensions
+     */
+    isValidFileExtension(fileName) {
+      const allowedExtensions = [".jpg", ".png"];
+      const fileExtension = fileName.slice(
+        ((fileName.lastIndexOf(".") - 1) >>> 0) + 2
+      );
+      return allowedExtensions.includes("." + fileExtension.toLowerCase());
```
Comment (fagubla) [inline resources/js/components/shared/ModalSaveVersion.vue:398]: return allowedExtensions.includes(fileName.slice(fileName.lastIndexOf(".")).toLowerCase());

File: resources/js/components/shared/ModalSaveVersion.vue [hunk 2]
```diff
@@ -492,44 +506,46 @@ export default {
...
+                this.hideModal();
+              })
+              .catch((error) => {
+                if (error.response.status && error.response.status === 422) {
+                  this.errors = error.response.data.errors;
+                }
+              });
+          })
+          .catch((err) => {
+            console.log(err);
```
Comment (fagubla) [inline resources/js/components/shared/ModalSaveVersion.vue:544]: use console.error

File: resources/js/components/shared/ModalSaveVersion.vue [hunk 3]
```diff
@@ -492,44 +506,46 @@ export default {
     },
     saveModal() {
       //if method is called from package-version PUBLISH button
-      if(this.origin !== "core") {
+      if (this.origin !== "core") {
```
Comment (fagubla) [inline resources/js/components/shared/ModalSaveVersion.vue:509]:
```
if (this.origin === "core") {
    this.saveFromEditLaunchpad();
    return;
} 
window.ProcessMaker.EventBus.$emit( 
. . .
```

**Review by fagubla — CHANGES_REQUESTED** [review]

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-686959223a.eng.processmaker.net

File: resources/js/components/shared/ModalSaveVersion.vue [hunk 4]
```diff
@@ -357,18 +349,29 @@ export default {
             return;
           }
 
-          Array.from(files).forEach((file) => {
+        this.validateImage(files);
```
[the remark this answers is not in the document]
Comment (CarliPinell) [reply resources/js/components/shared/ModalSaveVersion.vue:352]: solved

File: resources/js/components/shared/ModalSaveVersion.vue [hunk 5]
```diff
@@ -357,18 +349,29 @@ export default {
             return;
           }
 
-          Array.from(files).forEach((file) => {
+        this.validateImage(files);  
+        }
+      }
+    },
+    validateImage(files) {
```
[the remark this answers is not in the document]
Comment (CarliPinell) [reply resources/js/components/shared/ModalSaveVersion.vue:356]: solved

File: resources/js/components/shared/ModalSaveVersion.vue (hunk 1 above)
Comment (CarliPinell) [reply resources/js/components/shared/ModalSaveVersion.vue:398]: solved

File: resources/js/components/shared/ModalSaveVersion.vue (hunk 3 above)
Comment (CarliPinell) [reply resources/js/components/shared/ModalSaveVersion.vue:509]: solved

File: resources/js/components/shared/ModalSaveVersion.vue (hunk 2 above)
Comment (CarliPinell) [reply resources/js/components/shared/ModalSaveVersion.vue:544]: solved

File: resources/js/components/shared/ModalSaveVersion.vue [hunk 6]
```diff
@@ -357,18 +349,29 @@ export default {
...
+                  this.showDeleteIcons.push(false);
+                };
+                reader.readAsDataURL(file);
+              } else {
+                ProcessMaker.alert(
+                  this.$t("Only PNG and JPG extensions are allowed."),
+                  "danger"
+                );
+                return;
+              }
```
[the remark this answers is not in the document]
Comment (CarliPinell) [reply resources/js/components/shared/ModalSaveVersion.vue:372]: solved

File: resources/js/components/shared/ModalSaveVersion.vue [hunk 7]
```diff
@@ -360,20 +352,34 @@ export default {
...
-              };
-              reader.readAsDataURL(file);
-            }
-          });
+          this.validateImageExtension(files);
         }
       }
     },
+    /**
+     *
```
Comment (fagubla) [inline resources/js/components/shared/ModalSaveVersion.vue:360]: remove extra line

**Review by fagubla — CHANGES_REQUESTED** [review]

File: resources/js/components/shared/ModalSaveVersion.vue (hunk 7 above)
Comment (CarliPinell) [reply resources/js/components/shared/ModalSaveVersion.vue:360]: solved

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-686959223a.eng.processmaker.net

**Pull request merged**